lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -67,55 +65,170 @@ def scrub(transcript)
+          xml.pan             payment_method.number
+          xml.cvv             payment_method.verification_value
+          xml.expirationMonth format(payment_method.month, :two_digits)
+          xml.expirationYear  format(payment_method.year, :four_digits)
+        end
       end
 
-      def commit(action, parameters)
+      def add_debit(xml, money, options)
+        currency    = options[:currency] || currency(money)
+        description = options[:description].blank? ? 'Purchase' : options[:description]
+
+        xml.debit do
+          xml.transactionId new_transaction_id
+
+          add_customer_data(xml, options)
+
+          xml.amount      money
+          xml.currency    currency
+          xml.description description
+          xml.callbackUrl(options[:callback_url] || 'http://example.com')
+        end
+      end
+
+      def add_customer_data(xml, options)
+        # Ixopay returns an error if the elements are not added in the order used here.
+        xml.customer do
+          add_billing_address(xml,  options[:billing_address])  if options[:billing_address]
+          add_shipping_address(xml, options[:shipping_address]) if options[:shipping_address]
+
+          if options.dig(:billing_address, :company)
+            xml.company options[:billing_address][:company]
+          end
+
+          xml.email options[:email]
+          xml.ipAddress(options[:ip] || '127.0.0.1')
+        end
+      end
+
+      def add_billing_address(xml, address)
+        if address[:name]
+          xml.firstName split_names(address[:name])[0]
+          xml.lastName  split_names(address[:name])[1]
+        end
+
+        xml.billingAddress1 address[:address1]
+        xml.billingAddress2 address[:address2]
+        xml.billingCity     address[:city]
+        xml.billingPostcode address[:zip]
+        xml.billingState    address[:state]
+        xml.billingCountry  address[:country]
+        xml.billingPhone    address[:phone]
+      end
+
+      def add_shipping_address(xml, address)
+        if address[:name]
+          xml.shippingFirstName split_names(address[:name])[0]
+          xml.shippingLastName  split_names(address[:name])[1]
+        end
+
+        xml.shippingCompany   address[:company]
+        xml.shippingAddress1  address[:address1]
+        xml.shippingAddress2  address[:address2]
+        xml.shippingCity      address[:city]
+        xml.shippingPostcode  address[:zip]
+        xml.shippingState     address[:state]
+        xml.shippingCountry   address[:country]
+        xml.shippingPhone     address[:phone]
+      end
+
+      def new_transaction_id
+        SecureRandom.uuid
+      end
+
+      def commit(request)
         url = (test? ? test_url : live_url)
-        response = parse(ssl_post(url, post_data(action, parameters)))
+
+        begin
+          raw_response = ssl_post(url, request, headers(request))
+        rescue StandardError => error
+          return response_from_request_error(error)
+        end
```

Comment:
What is the intent of this code? I'm able to delete all of this out along with the `response_from_request_error` method below, and all of the tests still pass.